website/blog/2025-07-11-full-stack-observability-hyperdx-ferretdb.md

```diff
@@ -0,0 +1,201 @@
+- **Simplified infrastructure:** If your existing data infrastructure is already based on PostgreSQL, integrating HyperDX with FerretDB can streamline your database management and reduce operational overhead for your observability solution.
+- **No vendor lock-in:** Enjoy the freedom of truly open-source solutions without concerns about proprietary licensing or vendor lock-in.
+
+## Connecting HyperDX to FerretDB
+
+Connecting a self-hosted HyperDX instance to your FerretDB instance is straightforward, as HyperDX expects a MongoDB-compatible database.
+Here's a step-by-step guide to get you started:
+
+1. **Ensure FerretDB is running:** Make sure your FerretDB instance is active and accessible.
+   You can set it up with the [FerretDB evaluation image](https://docs.ferretdb.io/installation/evaluation/), by running the following command:
+
+   ```sh
+    docker run -d --name ferretdb -p 27017:27017 \
+    -e POSTGRES_USER=<username> \
+    -e POSTGRES_PASSWORD=<password> \
+    -v ./data:/var/lib/postgresql/data \
+    ghcr.io/ferretdb/ferretdb-eval:2
+   ```
+
+   Replace `<username>` and `<password>` with your desired authentication credentials.
+
+   If you want to explore more installation options, refer to our [FerretDB Installation Guide](https://docs.ferretdb.io/installation/).
+
+2. **Set up HyperDX:** You can run HyperDX as a standalone Docker container or as part of a Docker Compose setup.
+   For this guide, we'll use the Docker Compose setup provided by HyperDX, as it conveniently bundles all necessary components like the OpenTelemetry Collector and ClickHouse, allowing for a quick and comprehensive demonstration of data ingestion and visualization.
+
+   Start by cloning the HyperDX repository and switch to the `v2` branch, which contains the necessary configurations for self-hosting.
+
+   ```sh
+   git clone https://github.com/hyperdxio/hyperdx.git
+   cd hyperdx
+   # switch to the v2 branch
+   git checkout v2
+   ```
+
+3. **Configure HyperDX's database connection:** Before you start up HyperDX and other components, you'll typically replace the `mongo` service with your ferretdb service.
+   Update the `MONGO_URI` environment variable for HyperDX (with service named `app`) to point to your FerretDB instance.
+
+   ```yaml
+   environment:
+     - MONGO_URI=mongodb://<username>:<password>@ferretdb:27017/hyperdx
+   ```
+
+   Ensure to replace `<username>` and `<password>` with your FerretDB authentication details if enabled and that the container can access your FerretDB instance.
+   Comment out the MongoDB service (set as `db`) in the `docker-compose.yml` file, since you will be using FerretDB instead.
+   Since you are commenting out the MongoDB service (`db`), ensure to remove any references to it in the `docker-compose.yml` file, such as the `depends_on` section for the `app` service.
```

Comment:
@chilagrow This is the update on the dependence.

Reply:
Ok, good it's the same in your browser. Is it possible to crop with enough margin on top, or adjust width etc to display search bar upon zoom?

Reply:
Thanks for taking a look.